src/main/java/org/passay/NumberRangeRule.java

```diff
@@ -0,0 +1,135 @@
+/* See LICENSE for licensing and NOTICE for copyright. */
+package org.passay;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Rule for determining if a password contains any number within a defined range, lower inclusive, upper exclusive.
+ *
+ * @author  Middleware Services
+ */
+public class NumberRangeRule implements Rule
```

Comment:
I didn't want to dive into that rule because it's fairly complex, but you're correct. Refactoring `IllegalSequenceRule` to leverage StringMatch is the way to go.